lib/types/result-set.js

```diff
@@ -10,30 +10,76 @@ const asyncIteratorSymbol = Symbol.asyncIterator || "@@asyncIterator";
 
 /** @module types */
 
-// Old definition of the Result set
-// /**
-//  * Creates a new instance of ResultSet.
-//  * @class
-//  * @classdesc Represents the result of a query.
-//  * @param {Object} response
-//  * @param {String} host
-//  * @param {Object} triedHosts
-//  * @param {Number} speculativeExecutions
-//  * @param {Number} consistency
-//  * @param {Boolean} isSchemaInAgreement
-//  * @constructor
-//  */
-// function ResultSet(response, host, triedHosts, speculativeExecutions, consistency, isSchemaInAgreement) {
-
 /**
  * Represents the result of a query.
  */
 
 class ResultSet {
+    // Old definition of the Result set
+    // /**
+    //  * @param {Object} result
+    //  * @param {String} host
+    //  * @param {Object} triedHosts
+    //  * @param {Number} speculativeExecutions
+    //  * @param {Number} consistency
+    //  * @param {Boolean} isSchemaInAgreement
+    //  */
     /**
      * @param {rust.QueryResultWrapper} result
      */
-    constructor(result) {
+    constructor(
+        result,
+        host,
+        triedHosts,
+        speculativeExecutions,
+        consistency,
+        isSchemaInAgreement,
+    ) {
+        if (!(result instanceof rust.QueryResultWrapper)) {
+            console.warn(
+                "Using legacy version of the ResultSet constructor.\n" +
+                    "This is deprecated and may be removed at any moment.",
+            );
```

Comment:
Isn't there some customary way of issuing deprecation warnings in JS? In Rust, you would just use `#[deprecated]` attributes.

Reply:
(I'm OK with this way, just wondering if it's the usual way of doing this)

Reply:
I'm not aware of dedicated depreciation warning in JS.